doc/TESTING.md

```diff
@@ -1,22 +1,100 @@
 
-## Tooling
-Please explain what kind of tools you use for testing, why you have choosen them and how to work with them.
+In future other testing types like Integration Testing, Performance Testing can be added in the Continuous Integration process.
 
+## Why Do We Write Tests
+We write tests to verify that our software works as intended and to catch bugs and errors before they pushed and make it into production. Testing helps us catch issues early in the development process, which can save us time and resources in the long run. It also helps ensure that our software meets the requirements.
+
+## Our Goal in Testing
+Our goal is to write automated unit tests for each module or function in our codebase to ensure that it works as intended and the changes do not introduce regressions. The minimum code coverage of unit tests should be **70%**. To achieve this goal, we follow a comprehensive testing strategy of TDD (Test-driven development) process.
+
+## TDD (Test-driven development)
+**telemetry-functions** follows TDD (Test-driven development) process, a method of implementing software programming that interlaces unit testing, programming and refactoring on source code, to eusure the quality, reliability, and functionality of the software.
+
+Test-driven development (TDD) is a software development approach where developers write automated tests before writing the code. The idea is to write a test case that fails initially and then write the code to make the test pass. The process is repeated for each new feature or change to the codebase.
+
+## Tooling - Testing Libraries Installed
+**Jest** - a JavaScript testing framework is testing library used to write the test cases,  It allows you to write tests with an approachable, familiar and feature-rich API that gives you results quickly
+
+## How to install Jest
+Install the jest package (and optional typings) to the existing project's package.json by the bellow command
+
+npm install --save-dev jest @types/jest
+
+When cloning the project the package.json have the library in dev dependency which installs Jest with "npm install" in the project's root directoty.
 
 ## How to write tests
-This section should explain how to write different tests on a basic level. Please also include information about your testing strategies. Don't hesitate to use a lot of code exaples and links to further information and external tech documentation if you use .
+
+Jest is configured by default to look for .js, .jsx, .ts and .tsx files inside of __tests__ folders, as well as any files with a suffix of .test or .spec (this includes files called test or spec).
+
+Our project have __tests__ folders which have *.test.ts files that will contain our test cases and fixtures folder having all the payload mocks used in test cases.
```

Comment:
Switch to 3rd person ("the project has...") instead of 1st person ("Our project has...")